rust/scx_utils/src/topology.rs

```diff
@@ -692,10 +711,100 @@ fn is_smt_active() -> Option<bool> {
     Some(smt_on == 1)
 }
 
+fn replace_with_virt_llcs(node: &mut Node, min_cores: usize, max_cores: usize) -> Result<()> {
```

Comment:
I think adding more step-by-step comments on what's happening would make it easier to read this function.